### PR TITLE
Add section on UncheckedInput types

### LINKED ISSUE #4552: Clarify what are Checked and Unchecked Types and the difference between them.
We created checked and unchecked type for models in the schema file. We should have an explanation on what's the purpose of them.

We can clarify that “Unchecked” input types allow you to perform some operations that prisma considers “dangerous”, like directly writing foreign keys and we allow you to choose either a “safe” or an unchecked input type when doing operations like create, update.

### PR DESCRIPTION
**Pull request by TasinIshmam**
## Describe this PR

Adds a section on `UncheckedInput` types in the [Advanced type safety](https://docs-git-fork-tasinishmam-document-checked-unchec-9bc175-prisma.vercel.app/concepts/components/prisma-client/advanced-type-safety) page. 

## What issue does this fix?

Fixes #4552

### CODE REVIEW

**Review by nikolasburk — APPROVED** [review]

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx [hunk 1]
```diff
@@ -127,4 +127,109 @@ userPosts.posts
 userPosts.profile
 ```
 
-> See the [model query options](/reference/api-reference/prisma-client-reference#model-query-options) reference for more information about the different types available.
+> See the [model query options](../../../../reference/api-reference/prisma-client-reference#model-query-options) reference for more information about the different types available.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: Was this change necessary? The previous link should work just fine.

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx [hunk 2]
```diff
@@ -127,4 +127,109 @@ userPosts.posts
...
+
+### Generated <inlinecode>UncheckedInput</inlinecode> types
+
+The `UncheckedInput` types are a special set of generated types that allow you to perform some operations that Prisma Client considers "unsafe", like directly writing foreign keys. You can choose either the "safe" `Input` types or the "unsafe" `UncheckedInput` type when doing operations like `create`, `update`, or `upsert`.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: Are there other use cases than foreign keys?

Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: I think in Prisma context "foreign keys" are usually referred to as "relation fields", foreign keys are a database level concept.

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx [hunk 3]
```diff
@@ -127,4 +127,109 @@ userPosts.posts
...
+
+The first tab below shows an example Prisma schema with a one-to-many relation between `User` and `Post`. The second tab shows the `PostUncheckedCreateInput` generated type. It contains the `authorId` property, which is a foreign key. The third tab shows an example query that uses the `PostUncheckedCreateInput` type. This query will result in an error if a user with an `id` of `1` does not exist.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: I think the page would be easier to read if the first tab would be split out to stand by itself, and then the two tabs each for the Unchecked and Checked inputs beneath that.

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx [hunk 4]
```diff
@@ -127,4 +127,109 @@ userPosts.posts
...
+})
+```
+
+</tab>
+
+</TabbedContent>
+
+This query will also result in an error if an author with an `id` of `1` does not exist. In this case, Prisma Client will give a more descriptive error message. You can also use the [`connectOrCreate`](../../../../reference/api-reference/prisma-client-reference#connectorcreate) API to safely create a new user if one does not already exist with the given `id`.
+
+We recommend using the "safe" `Input` types whenever possible. However, there are some cases where you may need to use the "unsafe" `UncheckedInput` types.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: What are those use cases?

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx (hunk 2 above)
Comment (TasinIshmam) [reply content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: That's the only one I know, to be honest. I was quoting previous [discussions](https://prisma-company.slack.com/archives/C016KUHB1R6/p1636563619030800?thread_ts=1636563201.030700&cid=C016KUHB1R6) and [support issues](https://github.com/prisma/prisma/discussions/10121). 

Any suggestions on how to rephrase this?  @janpio

Comment (TasinIshmam) [reply content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: I think the relation field refers to the virtual fields that don't exist in the database (`author`, `posts`, etc.). Our docs do refer to the actual foreign keys as "foreign keys" here: https://www.prisma.io/docs/concepts/components/prisma-schema/relations. 

So perhaps this is fine?

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx (hunk 4 above)
Comment (TasinIshmam) [reply content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: This is incorrectly phrased. It's not as much an alternative use case as it is for an alternative syntax. I'll correct it. Thanks for picking this up

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx (hunk 2 above)
Comment (janpio) [reply content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: "relation scalar field" is the proper name for what is the foreign key on SQL/database level: https://www.prisma.io/docs/concepts/components/prisma-schema/relations#relation-scalar-fields

(Many of the uses of "foreign key" on that other page should be replaced by a "relation scalar field" probably)

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx [hunk 5]
```diff
@@ -128,3 +127,106 @@ userPosts.profile
...
+</TabbedContent>
+
+The same query can be rewritten using the "safer" `PostCreateInput` type. This type does not contain the `authorId` field but instead contains the `author` relation field.
+
+<TabbedContent tabs={[ <FileWithIcon text="Generated type" icon="code"/>, <FileWithIcon text="Example query" icon="code"/>]}>
+
+<tab>
+
+```ts
+type PostCreateInput = {
```
Comment (aqrln) [inline content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx:193]: I think the id is missing here:

```suggestion
type PostCreateInput = {
  id?: number
```

File: content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx [hunk 6]
```diff
@@ -128,3 +127,107 @@ userPosts.profile
...
+  id?: number
```
Comment (TasinIshmam) [inline content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]: Hey, I just checked and this is not generated inside `PostCreateInput`. Only the unchecked input type `PostUncheckedCreateInput` allows you to specify the `id` field directly.  

@ruheni

Comment (ruheni) [inline content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx]:
```suggestion
```
